src/3rd_party/topk.cuh

```diff
@@ -0,0 +1,302 @@
+// element.cu is removed. When this flag is true, the indices returns are the offsets within the row. When the flag is false, the indices
+// returned are offset from the base pointer.
+template <typename IndexType, typename T, bool getRowOffsets=false>
+void topK_kernelLauncher(T* log_probs,
+                         IndexType* topk_tmp_id_buf,
+                         T* topk_tmp_val_buf,
+                         TopK<IndexType, T>* tops,
+                         const int batch_size,
+                         const int beams_per_batch,
+                         const int k,
+                         const int vocab_size,
+                         bool descendingOrder,
+                         cudaStream_t stream) {
+  
+  IndexType* outIndices = nullptr;
+  T* outVals = nullptr;                        
+  switch(k) {
+    CASE_K(1,128,128,MAX_BLOCKS_PER_BEAM);
+    CASE_K(2,128,128,MAX_BLOCKS_PER_BEAM);
+    CASE_K(4,128,128,MAX_BLOCKS_PER_BEAM);
+    CASE_K(6,128,128,MAX_BLOCKS_PER_BEAM);
+    CASE_K(8,128,128,MAX_BLOCKS_PER_BEAM);
+    CASE_K(10,128,128,MAX_BLOCKS_PER_BEAM);
+    CASE_K(16,128,128,5);
+    CASE_K(32,256,128,1);
+    CASE_K(64,256,256,1);
+    default:
+      topk_stage_1<IndexType, T, 128, 1, getRowOffsets><<<batch_size * beams_per_batch * 1, 128, 0, stream>>>(log_probs, 
+                                                                                               topk_tmp_id_buf, 
+                                                                                               topk_tmp_val_buf,
+                                                                                               k, 
+                                                                                               vocab_size, 
+                                                                                               descendingOrder);
+
+      topk_stage_2<IndexType, T, 128, 1><<<batch_size, 128, k * sizeof(TopK<IndexType, T>), stream>>>(topk_tmp_id_buf,
+                                                                                                      topk_tmp_val_buf,
+                                                                                                      tops,
+                                                                                                      outIndices,
+                                                                                                      outVals,
+                                                                                                      beams_per_batch,
+                                                                                                      k,
+                                                                                                      descendingOrder);
+    break;
+  }
+}
+
+template <typename IndexType, typename T, bool getRowOffsets=false>
```

Comment:
Please add a function-level comment what htis does, and how it differs from the previous overload above.